File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
//
// 0,3,7,2,5,8,4,6,0,1 0 -- 8 == 9
//           2 
//
//
// N + M
// logic
// start counter = 1
// hash[100] = 1
// hash[99] or [101] > 1
// if [99] or [101] is <2, i increase the counter
// then counter++ afterward
// 
// hash value that is greater than 1 is not countable
var longestConsecutive = function(nums) {
    let longest = 0
    let set = new Set(nums)
    
    for (let num of set){
        if(!set.has(num - 1)){
            let current = 1
            let currentNum = num
            
            while(set.has(currentNum + 1)){
                current++
                currentNum++
            }
            longest = Math.max(longest, current)
        }
    }
    return longest
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest = 0
        num_set = set(nums)
        
        for num in num_set:
            print(num)
            if num - 1 not in num_set: ## if this is the first number of the streak
                current_num = num ## set the starting
                current = 1 ## set the starting streak
                
                while current_num + 1 in num_set: ## keep checking the next number
                    current_num += 1 ##increase number
                    current += 1 ##increase streak
                    
                longest = max(longest,current)## check if current streak is greater
                
        return longest